app/web_hub_test.go

```diff
@@ -106,3 +107,114 @@ func TestHubStopRaceCondition(t *testing.T) {
 		require.FailNow(t, "hub call did not return within 15 seconds after stop")
 	}
 }
+
+func TestHubConnIndex(t *testing.T) {
+	th := Setup(t).InitBasic()
+	defer th.TearDown()
+
+	connIndex := newHubConnectionIndex()
+
+	// User1
+	wc1 := &WebConn{
+		App:    th.App,
+		UserId: model.NewId(),
+	}
+
+	// User2
+	wc2 := &WebConn{
+		App:    th.App,
+		UserId: model.NewId(),
+	}
+	wc3 := &WebConn{
+		App:    th.App,
+		UserId: wc2.UserId,
+	}
+	wc4 := &WebConn{
+		App:    th.App,
+		UserId: wc2.UserId,
+	}
+
+	connIndex.Add(wc1)
+	connIndex.Add(wc2)
+	connIndex.Add(wc3)
+	connIndex.Add(wc4)
+
+	assert.True(t, connIndex.Has(wc1))
+	assert.True(t, connIndex.Has(wc2))
+
+	assert.ElementsMatch(t, connIndex.ForUser(wc2.UserId), []*WebConn{wc2, wc3, wc4})
+	assert.ElementsMatch(t, connIndex.ForUser(wc1.UserId), []*WebConn{wc1})
+	assert.True(t, connIndex.Has(wc2))
+	assert.True(t, connIndex.Has(wc1))
+	assert.Len(t, connIndex.All(), 4)
+
+	connIndex.Remove(wc3) // Remove from middle from user2
```

Comment:
Would it make sense to run tests with `t.Run`?

Reply:
done